repokid/role.py

```diff
@@ -0,0 +1,61 @@
+class Role(object):
+    def __init__(self, role_dict):
+        self.aa_data = role_dict.get('AAData', {})
+        self.active = role_dict.get('Active', True)
+        self.arn = role_dict.get('Arn', None)
+        self.assume_role_policy_document = role_dict.get('AssumeRolePolicyDocument', None)
+        self.create_date = role_dict.get('CreateDate', None)
+        self.disqualified_by = role_dict.get('DisqualifiedBy', [])
+        self.policies = role_dict.get('Policies', [])
+        self.repoable_permissions = role_dict.get('RepoablePermissions', 0)
+        self.repoable_services = role_dict.get('RepoableServices', [])
+        self.repoed = role_dict.get('Repoed', '')
+        self.role_id = role_dict.get('RoleId', None)
+        self.role_name = role_dict.get('RoleName', None)
+        self.stats = role_dict.get('Stats', [])
+        self.total_permissions = role_dict.get('TotalPermissions', 0)
+
+        self.account = role_dict.get('Account', None) or self.arn.split(':')[4] if self.arn else None
+
+    def as_dict(self):
+        return {'AAData': getattr(self, 'aa_data', None),
+                'Account': getattr(self, 'account', None),
+                'Active': getattr(self, 'active', None),
+                'Arn': getattr(self, 'arn', None),
+                'AssumeRolePolicyDocument': getattr(self, 'assume_role_policy_document', None),
+                'CreateDate': getattr(self, 'create_date', None),
+                'DisqualifiedBy': getattr(self, 'disqualified_by', None),
+                'Policies': getattr(self, 'policies', None),
+                'RepoablePermissions': getattr(self, 'repoable_permissions', None),
+                'RepoableServices': getattr(self, 'repoable_services', None),
+                'Repoed': getattr(self, 'repoed', None),
+                'RoleId': getattr(self, 'role_id', None),
+                'RoleName': getattr(self, 'role_name', None),
+                'Stats': getattr(self, 'stats', None),
+                'TotalPermissions': getattr(self, 'total_permissions', None)}
+
+
+class Roles(object):
+    def __init__(self, role_object_list):
+        self.roles = role_object_list
+
+    def __getitem__(self, index):
+        return self.roles[index]
+
+    def __len__(self):
+        return len(self.roles)
+
+    def role_id_list(self):
+        return [role.role_id for role in self.roles]
+
+    def get_by_id(self, id):
+        try:
+            return self.filter(role_id=id)[0]
+        except IndexError:
+            return None
+
+    def filter(self, *args, **kwargs):
+        roles = self.roles
+        for arg, value in kwargs.items():
+            roles = [role for role in roles if getattr(role, arg, None) == value]
```

Comment:
Can't use role.get because it isn't a dictionary, but I can just do self.xxx since I set defaults in the constructor.  Will change.